packages/dd-trace/test/debugger/devtools_client/condition.spec.js

```diff
@@ -0,0 +1,261 @@
+  [{ index: [{ ref: 'obj' }, 'bar'] }, { obj: { foo: 'bar' } }, undefined], // Should throw according to spec
+  [
+    { index: [{ ref: 'set' }, 'foo'] },
+    { set: new Set(['foo']) },
+    new Error('Aborting because accessing a Set or WeakSet is not allowed')
+  ],
+  [
+    { index: [{ ref: 'set' }, 'bar'] },
+    { set: new Set(['foo']) },
+    new Error('Aborting because accessing a Set or WeakSet is not allowed')
+  ],
+  [{ index: [{ ref: 'map' }, 'foo'] }, { map: new Map([['foo', 'bar']]) }, 'bar'],
+  [{ index: [{ ref: 'map' }, 'bar'] }, { map: new Map([['foo', 'bar']]) }, undefined], // Should throw according to spec
+  [{ index: [{ ref: 'wmap' }, { ref: 'key' }] }, { key: weakKey, wmap: new WeakMap([[weakKey, 'bar']]) }, 'bar'],
+  [
+    { index: [{ ref: 'wmap' }, { ref: 'key' }] },
+    { key: {}, wmap: new WeakMap([[weakKey, 'bar']]) },
+    undefined // Should throw according to spec
+  ],
+  [
+    { index: [{ ref: 'set' }, { ref: 'key' }] },
+    { key: weakKey, set: new WeakSet([weakKey]) },
+    new Error('Aborting because accessing a Set or WeakSet is not allowed')
+  ],
+  [
+    { index: [{ ref: 'set' }, { ref: 'key' }] },
+    { key: {}, set: new WeakSet([weakKey]) },
+    new Error('Aborting because accessing a Set or WeakSet is not allowed')
+  ],
+  [
+    { index: [{ ref: 'obj' }, 'getter'] },
+    { obj: Object.create(Object.prototype, { getter: { get () { return 'x' } } }) },
+    new Error('Aborting because of possible side effects')
+  ],
+
+  // Argument predicates and operations
+  [{ eq: [{ ref: 'hits' }, true] }, { hits: true }, true],
+  [{ eq: [{ ref: 'hits' }, null] }, { hits: null }, true],
+  [{ substring: [{ ref: 'payload' }, 4, 7] }, { payload: 'hello world' }, 'hello world'.slice(4, 7)],
+  [{ any: [{ ref: 'collection' }, { isEmpty: { ref: '@it' } }] }, { collection: ['foo', 'bar', ''] }, true],
+  [{ any: [{ ref: 'coll' }, { isEmpty: { ref: '@value' } }] }, { coll: { 0: 'foo', 1: 'bar', 2: '' } }, true],
+  [{ any: [{ ref: 'coll' }, { isEmpty: { ref: '@value' } }] }, { coll: { 0: 'foo', 1: 'bar', 2: 'baz' } }, false],
+  [{ any: [{ ref: 'coll' }, { isEmpty: { ref: '@key' } }] }, { coll: { foo: 0, bar: 1, '': 2 } }, true],
+  [{ any: [{ ref: 'coll' }, { isEmpty: { ref: '@key' } }] }, { coll: { foo: 0, bar: 1, baz: 2 } }, false],
+  [{ startsWith: [{ ref: 'local_string' }, 'hello'] }, { local_string: 'hello world!' }, true],
+  [{ startsWith: [{ ref: 'local_string' }, 'world'] }, { local_string: 'hello world!' }, false],
+  [
+    { filter: [{ ref: 'collection' }, { not: { isEmpty: { ref: '@it' } } }] },
+    { collection: ['foo', 'bar', ''] },
+    ['foo', 'bar']
+  ],
+  [
+    { filter: [{ ref: 'collection' }, { not: { isEmpty: { ref: '@it' } } }] },
+    { collection: new Set(['foo', 'bar', '']) },
+    ['foo', 'bar']
+  ],
+  [
+    { filter: [{ ref: 'collection' }, { not: { isEmpty: { ref: '@value' } } }] },
+    { collection: { 1: 'foo', 2: 'bar', 3: '' } },
```

Comment:
does the spec allow the user to have '@it' iterator over entities? not sure if that something we need to support..

Reply:
I've been wondering the same thing. I came to the conclusion that shouldn't be supported based on [this discussion](https://docs.google.com/document/d/1g76oXz9A5NV4y8Do8eSwmUXOKdZxVE61kbQSJGpBZ-E/edit?tab=t.0#bookmark=id.jtx0oaxna0rj) from January, but maybe I misunderstood it.

Reply:
Consensus from Zoom: We'll keep the current implementation